xmtp_mls/src/groups/members.rs

```diff
@@ -48,19 +53,28 @@ impl MlsGroup {
             .collect::<Vec<_>>();
 
         let conn = provider.conn_ref();
-        let association_states =
+        let mut association_states =
             StoredAssociationState::batch_read_from_cache(conn, requests.clone())?;
         let mutable_metadata = self.mutable_metadata(provider)?;
         if association_states.len() != requests.len() {
-            // Cache miss - not expected to happen because:
-            // 1. We don't allow updates to the group metadata unless we have already validated the association state
-            // 2. When validating the association state, we must have written it to the cache
-            log::error!(
-                "Failed to load all members for group - metadata: {:?}, computed members: {:?}",
-                requests,
-                association_states
-            );
-            return Err(GroupError::InvalidGroupMembership);
+            // Attempt to rebuild the cache before erroring out due to a cache miss.
+            let requests: Vec<_> = requests
+                .into_iter()
+                .map(|(id, sequence)| (id, Some(sequence)))
```

Comment:
In the narrow case of your PR, this is fine. But really we should only be doing these lookups for items that are missing from the cache.

Reply:
Got it. Filtering those out now.